src/App.tsx

```diff
@@ -1,56 +1,79 @@
-import React from 'react';
 import './App.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import cn from 'classnames';
+import { debounce } from 'lodash';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './Components/DropdownMenu';
+
+const TIME_DELAY = 1000;
+
+function filterPeople(peopleToFilter: Person[], query: string) {
+  const transformedQuery = query.toLowerCase().trim();
+
+  return peopleToFilter
+    .filter(({ name }) => name.toLocaleLowerCase().includes(transformedQuery));
```

Comment:
You use `toLowerCase` and `toLocaleLowerCase` - that potentially may work wrong
I suggest you using same method in both cases